app/src/ereferrals/java/org/eyeseetea/malariacare/strategies/DashboardActivityStrategy.java

```diff
@@ -526,12 +557,69 @@ public void reloadAVFragment() {
 
     public void showEndSurveyMessage(SurveyDB surveyDB) {
         if (surveyDB != null && !noIssueVoucher(surveyDB) && !hasPhone(surveyDB)) {
-            mDashboardActivity.showException("", String.format(
-                    mDashboardActivity.getResources().getString(R.string.give_voucher),
-                    surveyDB.getEventUid()));
+            final String voucherUId = surveyDB.getEventUid();
+
+            GetSettingsUseCase getSettingsUseCase = new GetSettingsUseCase(new UIThreadExecutor(), new AsyncExecutor(),
+                    new SettingsDataSource(mDashboardActivity.getBaseContext()));
+            getSettingsUseCase.execute(new GetSettingsUseCase.Callback() {
+                @Override
+                public void onSuccess(Settings setting) {
+                    DialogInterface.OnClickListener onClickListener = null;
+                    if(setting.isElementActive()){
+                        onClickListener = createOnClickListenerToSendVoucherToElement(voucherUId);
+                    }
+
+                    mDashboardActivity.showException(mDashboardActivity, "", String.format(
+                            mDashboardActivity.getResources().getString(R.string.give_voucher),
+                            voucherUId), onClickListener);
+                }
+            });
         }
     }
 
+    @NonNull
+    private DialogInterface.OnClickListener createOnClickListenerToSendVoucherToElement(final String voucherUId) {
```

Comment:
There is a lot of Element-dependent logic in the dashboard activity. It's probably the same "naming-convention" problem, but I would expect the dashboard to talk about post-actions that are needed, not about a concrete technology.